pkg/cloudprovider/fake/factory.go

```diff
@@ -28,13 +27,16 @@ var (
 
 type Factory struct {
 	WantErr error
-	// NodeReplicas is use by tests to control observed replicas.
-	NodeReplicas map[string]int32
+	// NodeReplicas is used by tests to control observed replicas.
+	NodeReplicas     map[string]*int32
+	NodeGroupStable  bool
+	NodeGroupMessage string
```

Comment:
Consider embedding a fake error message into the fake.Factory implementation so that tests don't need to configure it.

Maybe even make a public const that tests can reference

const ErrorMessage = "fake factory failed"

Reply:
Are you suggesting doing something like this- 

```
const (
	ErrMessage = ""
)

func NewFactory(options cloudprovider.Options) *Factory {
	return &Factory{
		NodeReplicas:     make(map[string]*int32),
		NodeGroupStable:  true,
		NodeGroupMessage: ErrMessage,
	}
}
```

and in tests we will be doing this - 
```
fake.ErrMessage = dummyMessage
```

I am not sure how does this help?

Reply:
In the test we wouldn't need to do `fake.ErrMessage = dummyMessage`. You would only return the ErrMessage if NodeGroupStable is false. Either way works, but the former results in less test code.